homeassistant/components/samsungtv/media_player.py

```diff
@@ -179,21 +193,72 @@ async def async_update(self) -> None:
-            await self._bridge.async_request_app_list()
-            if self._app_list_event.is_set():
-                # The try+wait_for is a bit expensive so we should try not to
-                # enter it unless we have to (Python 3.11 will have zero cost try)
-                return
-            try:
-                await asyncio.wait_for(self._app_list_event.wait(), APP_LIST_DELAY)
-            except asyncio.TimeoutError as err:
-                # No need to try again
-                self._app_list_event.set()
-                LOGGER.debug(
-                    "Failed to load app list from %s: %s", self._host, err.__repr__()
+        if self._attr_state == STATE_ON:
+            startup_tasks: list[Coroutine[Any, Any, None]] = []
+
+            if not self._app_list_event.is_set():
+                startup_tasks.append(self._async_startup_app_list())
+
+            if not self._upnp_device:
+                startup_tasks.append(self._async_startup_upnp())
+
+            await asyncio.gather(*startup_tasks)
+
+            if service := self._get_upnp_service():
+                action = service.action("GetVolume")
+                get_volume = await action.async_call(InstanceID=0, Channel="Master")
+                LOGGER.debug("Upnp GetVolume on %s: %s", self._host, get_volume)
+                if (volume_level := get_volume.get("CurrentVolume")) is not None:
+                    self._attr_volume_level = volume_level / 100
+
+                action = service.action("GetMute")
+                get_mute = await action.async_call(InstanceID=0, Channel="Master")
+                LOGGER.debug("Upnp GetMute on %s: %s", self._host, get_mute)
+                if (is_muted := get_mute.get("CurrentMute")) is not None:
+                    self._attr_is_volume_muted = is_muted
+
+    async def _async_startup_app_list(self) -> None:
+        await self._bridge.async_request_app_list()
+        if self._app_list_event.is_set():
+            # The try+wait_for is a bit expensive so we should try not to
+            # enter it unless we have to (Python 3.11 will have zero cost try)
+            return
+        try:
+            await asyncio.wait_for(self._app_list_event.wait(), APP_LIST_DELAY)
+        except asyncio.TimeoutError as err:
+            # No need to try again
+            self._app_list_event.set()
+            LOGGER.debug(
+                "Failed to load app list from %s: %s", self._host, err.__repr__()
+            )
+
+    async def _async_startup_upnp(self) -> None:
+        if self._upnp_device is None:
+            session = async_get_clientsession(self.hass)
+            upnp_requester = AiohttpSessionRequester(session)
+            upnp_factory = UpnpFactory(upnp_requester)
+            with contextlib.suppress(UpnpConnectionError):
+                self._upnp_device = await upnp_factory.async_create_device(
+                    f"http://{self._host}:9197/dmr"
```

Comment:
We should be able to get the port from `async_step_ssdp`

Reply:
I think we also need to shutdown the `self._upnp_device` in `async_will_remove_from_hass`